fw/cache.c

```diff
@@ -178,15 +178,20 @@ __tfw_dbg_dump_ce(const TfwCacheEntry *ce)
 #define __tfw_dbg_dump_ce(...)
 #endif
 
+#define TFW_CSTR_DUPLICATE	TFW_STR_DUPLICATE
+#define TFW_CSTR_SPEC_IDX	0x2
+
 /**
  * String header for cache entries used for TfwStr serialization.
  *
- * @flags	- only TFW_STR_DUPLICATE or zero;
+ * @flags	- only TFW_CSTR_DUPLICATE and TFW_CSTR_HPACK_IDX or zero;
  * @len		- string length or number of duplicates;
+ * @idx		- HPACK static index or index of special header;
  */
 typedef struct {
 	unsigned long	flags : 8,
-			len : 56;
+			len : 48,
+			idx : 8;
```

Comment:
As for me, we can use `unsigned int` here. In this case `len` becomes 2 bytes and it's enough to store value of the header. If so for consistency we must introduce such limit to non-cacheable responses. I believe adding limits may  decrease risk of resource exhaustion attacks or overflow attacks.